Launch flow - Button (launch flow)

ihor-palii opened the pull request:
## Feature
User hit the Launch flow button and a new box will appear with
 - Start flow immediately and send to selected groups or individual
 - Start flow when a specified keyword is received in a message
 - Start flow at a future date and time

Details on: 
https://app.asana.com/0/1159876841343611/1158809372498669/f

teehamaral commented on templates/flows/flow_launch.haml:
```diff
@@ -0,0 +1,212 @@
+                  -trans "Interrupt contacts currently active in flows"
+    
+    #id_launch_on_keyword_trigger
+      - render_field 'keyword'
```
This field should have select2 as `tag` type as we have on `groups` field

ihor-palii replied:
Have this similar to create keyword trigger page.

teehamaral replied:
It should have the select2 instance on `extra_js` as well

ihor-palii replied:
Have added `select2` instance on `extra-script`

teehamaral replied:
The type of select2 should be as `tags` format like this https://github.com/greatnonprofits-nfp/rapidpro/blob/release/0.22/templates/flows/flow_launch_keyword.haml#L137

![Apr-16-2020 11-39-27](https://user-images.githubusercontent.com/5198870/79470649-0dd5cb80-7fd8-11ea-909d-6582d0996796.gif)

ihor-palii commented:
Ok, thanks. I'll take care of it.

ihor-palii commented on temba/flows/views.py:
```diff
@@ -83,6 +89,15 @@
+LAUNCH_ON_KEYWORD_TRIGGER = 'LAUNCH_ON_KEYWORD_TRIGGER'
+LAUNCH_ON_SHEDULE_TRIGGER = 'LAUNCH_ON_SHEDULE_TRIGGER'
+LAUNCH_CHOICES = (
+    (LAUNCH_IMMEDIATELY, 'Start flow immediately and send to selected groups or individual'),
```
Done

teehamaral commented on temba/flows/models.py:
```diff
@@ -1017,35 +1014,6 @@ def get_versions_after(cls, version_number):
     def as_select2(self):
         return dict(id=self.uuid, text=self.name)
 
-    def release(self):
```
Don't we need any more of this method?

ihor-palii replied:
We don't need, because this method is overwritten by 
https://github.com/greatnonprofits-nfp/rapidpro/blob/32b86a5ff78128310500512a51374c687d2ee4a8/temba/flows/models.py#L2133-L2164

ihor-palii commented:
> I added two comments and caught this little issue when I switched back to the option start immediately, I can see both of fields `recipients, contacts...` and `enter a query`
> 
> ![Apr-14-2020 12-26-43](https://user-images.githubusercontent.com/5198870/79243016-72175480-7e4b-11ea-92d9-f40f07e66a1f.gif)

This bug fixed.

teehamaral commented on templates/flows/flow_launch.haml:
```diff
@@ -0,0 +1,267 @@
+            minDate.setMinutes(0);
+            minDate.setHours(minDate.getHours() + 1);
+        }
+        initializeDatetimePicker(minDate, undefined, false);
```
The appearance of this `datetime` picker should be like this attached screenshot and you can find references here https://github.com/greatnonprofits-nfp/rapidpro/blob/release/0.22/templates/flows/flow_launch_schedule.haml and here 
https://github.com/greatnonprofits-nfp/rapidpro/commit/09be08924731f7837001c09f08ced9f690736a58#diff-2cb5ce8cbba47341d8f7a115565fdc1f

![Screen Shot 2020-04-16 at 12 23 06](https://user-images.githubusercontent.com/5198870/79475159-b4709b00-7fdd-11ea-8677-3bd3fa57d9e2.png)

ihor-palii commented:
Added requested changes
1. Changed calendar type
2. `select2` for `keyword_triggers` field.
3. Removed `group` and  `type` fields.
